core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2551,6 +2551,7 @@ private[util] object CallerContext extends Logging {
  */
 private[spark] class CallerContext(
    from: String,
+   upstreamCallerContext: Option[String] = None,
```

Comment:
It is possible to use named parameters, but I agree with the point of private scope + frequency of use here to be mitigating factor.